### DISCUSSION TITLE
pb_hooks directory

### DISCUSSION THREAD

**Discussion by charbs-io** [pockethost/pockethost#271] [Q&A] [answered]
Hi, 

I'm experimenting with PB's JS hooks released with [Pocketbase 0.17.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.17.0). 

I don't seem to be able to create a `pb_hooks` folder via FTP to add some JS files into.

Can someone shed some light on how I can create folders?

Thanks!

**Comment by benallfree** [comment]
@charbs-io JS hooks are not yet supported by PocketHost. Can you share what you're trying to accomplish?

**Comment by benallfree** [comment]
PS, would you like to submit a PR for this feature?

**Comment by charbs-io** [comment]
@benallfree i'll have a go sure. 

The context is i've got a simple app with some JSVM functions (hooks, some custom routes) and the SDK is expecting some of these routes to handle some specific backend functionality. All i need is a pb_hooks folder to drop some `.pb.js` files in and i'm set :)

**Comment by benallfree** [comment]
@charbs-io You'll probably want to start by looking at the FTP's virtual file system provider. Let me know what you think!

**Reply by charbs-io** [reply]
I am unfamiliar with a quite a few things on this repo :) 

Is `scripts/dev.sh` my starting point for local development @benallfree ?

**Reply by benallfree** [reply]
Yes. see https://pockethost.gitbook.io/manual/contributing/index

**Reply by charbs-io** [reply]
@benallfree insanely cool to be able to run the entire stack locally. Got it all running successfully. PR incoming. It may be too good to be true, but all I had to do really was ensure the FtpService was creating a new `pb_hooks` folder. PB handles all the rest of it (tested it with a few hooks.)

**Reply by charbs-io** [reply]
@benallfree are you able to give me write access? Doesn't look like i have the ability to push my local branch...

**Reply by benallfree** [reply]
Try submitting a pull request instead

**Reply by charbs-io** [reply]
PR Open, but i can't figure out how to assign you as a reviewer @benallfree (Shocking, but I don't use Github for any of my work...)

https://github.com/benallfree/pockethost/pull/272

**Comment by benallfree** [comment, marked as answer]
For anyone watching, a pb_hooks update has been quietly pushed for beta testing. Thanks @charbs-io for PR #272!

**Comment by HEAVYPOLY** [comment]
On this note, how do we do the equivalent with go extension?  https://pocketbase.io/docs/go-overview/

**Comment by benallfree** [comment]
@HEAVYPOLY PocketHost does not support custom builds of PocketBase. There are no plans to add support for custom-compiled versions.
